libs/agora/gene-comparison-tool/src/lib/gene-comparison-tool.component.ts

```diff
@@ -172,27 +181,32 @@ export class GeneComparisonToolComponent implements OnInit, AfterViewInit, OnDes
   @ViewChild('scorePanel') scorePanel!: ScorePanelComponent;
   @ViewChild('pinnedGenesModal') pinnedGenesModal!: PinnedGenesModalComponent;
 
-  constructor() {
-    this.category = 'RNA - Differential Expression';
-  }
-
   ngOnInit() {
-    this.urlParamsSubscription = this.route.queryParams.subscribe((params) => {
-      this.urlParams = params || {};
+    this.category = this.categoryParam ? this.categoryParam : this.categories[0].value;
 
-      this.category = this.urlParams['category'] || this.categories[0].value;
-      this.subCategory = this.urlParams['subCategory'] || '';
-      this.updateSubCategories();
+    if (this.subCategoryParam) {
+      this.subCategory = this.subCategoryParam;
+    }
+    this.updateSubCategories();
 
-      this.sortField = this.urlParams['sortField'] || '';
-      this.sortOrder = '1' === this.urlParams['sortOrder'] ? 1 : -1;
+    if (this.sortFieldParam) {
+      this.sortField = this.sortFieldParam;
+    }
 
-      this.significanceThreshold =
-        this.urlParams['significance'] || this.DEFAULT_SIGNIFICANCE_THRESHOLD;
-      this.significanceThresholdActive = !!this.urlParams['significance'];
+    if (this.sortOrderParam) {
+      this.sortOrder = this.sortOrderParam === '1' ? 1 : -1;
+    }
 
-      this.loadGenes();
-    });
+    if (this.significanceParam) {
+      const parsedValue = parseFloat(this.significanceParam);
+      if (!isNaN(parsedValue)) {
+        this.significanceThreshold = parsedValue;
+      }
+    }
+
+    this.significanceThresholdActive = !!this.significanceParam;
```

Comment:
Should `significanceThreshold` continue to fall back to `this.DEFAULT_SIGNIFICANCE_THRESHOLD`?  

Should `significanceThresholdActive` depend on whether `significanceThreshold` was set using the value of `significanceParam`?

```suggestion
  // default values
   this.significanceThreshold = this.DEFAULT_SIGNIFICANCE_THRESHOLD;
   this.significanceThresholdActive = false;
   
   // update if possible using param
    if (this.significanceParam) {
      const parsedValue = parseFloat(this.significanceParam);
      if (!isNaN(parsedValue)) {
        this.significanceThreshold = parsedValue;
        this.significanceThresholdActive = true;
      }
    } 
```

Reply:
I think the original code used the following line to handle the active state. I do think just setting the boolean would be clearer though since I am checking for a valid number.
    this.significanceThresholdActive = !!this.significanceParam;